lib/salus/scanners/base.rb

```diff
@@ -76,11 +79,24 @@ def run!(salus_report:, required:, pass_on_raise:, reraise:)
       salus_report.add_scan_report(@report, required: required)
 
       begin
-        @report.record { run }
+        @report.record do
+          Timeout.timeout(max_lifespan / 1000.to_f) { run }
+        end
 
         if @report.errors.any?
           pass_on_raise ? @report.pass : @report.fail
         end
+      rescue Timeout::Error => e
+        timeout_error_data = {
+          message: "Scanner #{name} timed out during execution",
+          error_class: ScannerTimeoutError,
+          backtrace: e.backtrace.take(5)
+        }
+        @report.error(timeout_error_data)
+        salus_report.error(timeout_error_data)
+
```

Comment:
I'm not sure if `error` is a callable method from `@report`/`salus_report`?

Reply:
Looks to be defined [here in report.rb](https://github.com/coinbase/salus/blob/master/lib/salus/report.rb#L72).